### ISSUE TITLE
[question] 

### ISSUE THREAD

**Issue by tschrijvers1999** [mui/mui-x#13207]
### Order ID or Support key 💳

89694

### Search keywords

colspan, leaffield

### Latest version

- [X] I have tested the latest version

### The problem in depth

I was wondering if it is possible to edit the leaffield. I build the following MUI grid, however I would like to use the space below part (which is a group) to show tasks:

![image](https://github.com/mui/mui-x/assets/148943133/5c255de6-0249-4db9-854c-6c44a8d640d1)

Part and a few other columns are also pinned to the left.

I have a few options in mind to save some space on the left and show the full part name:
- Is there an option to edit the leaffield?
- Can we reduce the grouped column (phase), but still show the full name (using colspan maybe)? This is closely related to the following issue, however this did not work for me: https://github.com/mui/mui-x/issues/6462

Other suggestions are also welcome, let me know if you can help me!

### Your environment

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by tschrijvers1999** [comment]
@michelengelen yes, however I cannot edit this field. Which I would like to do, otherwise it is not useful for me to have it there.

**Comment by michelengelen** [comment]
Yes, that is currently not possible.
@joserodolfofreitas is there a feature planned to render a custom leaffield, or just use the leaffield as is from the coldef?

**Comment by tschrijvers1999** [comment]
@michelengelen using colspan for the pinned group is also not possible? Since otherwise the column for the grouped parts needs to be really big, which causes a lot of useless whitespace.

**Comment by flaviendelangle** [comment]
> @michelengelen yes, however I cannot edit this field

The main problem here is that you most likely want every row inside this group to have their data updated for the edited field.
And we don't support it for now, we probably could in the future, but for me this is the main limiting factor here.

**Comment by michelengelen** [comment]
I don't think so ... I will test this tomorrow when I am back at my desk! 👍🏼

**Comment by michelengelen** [comment]
@tschrijvers1999 I tried today, but couldn't make it work.

@romgrk do you have any idea if this is possible in some way?

Otherwise I would consider this a feature request. Shouldn't be that hard to allow for editing in the leaffield. Would that be fine @tschrijvers1999?

**Comment by tschrijvers1999** [comment]
@michelengelen Thanks for looking into it.

If another solution is not possible than a feature request is fine. How long does such a request usually take before it is processed?

**Comment by michelengelen** [comment]
It really depends on the complexity and the workload the team is experiencing. 
To speed things up you can also pick this up yourself and open a PR with a solution. 👍🏼

**Comment by tschrijvers1999** [comment]
I do not have enough experiense, so I will wait for the new feature and workaround with an extra column for now.

**Comment by romgrk** [comment]
It should be possible to use the `renderCell` field to customize the logic.

**Comment by tschrijvers1999** [comment]
@romgrk do you have an example of how this should work? I guess I should than create a normal column and in rendercell make some logic such that it appears underneath the group (as a leaf field)?

**Comment by romgrk** [comment]
Documentation here: https://mui.com/x/react-data-grid/row-grouping/#grouping-rows-with-custom-cell-renderer
